sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -185,10 +185,15 @@ class OrcFileFormat
             isCaseSensitive, dataSchema, requiredSchema, reader, conf)
         }
 
-      if (requestedColIdsOrEmptyFile.isEmpty) {
+      if (requestedColIdsOrEmptyFile._2) {
+        resultSchemaString = OrcUtils.orcTypeDescriptionString(actualSchema)
+      }
+      OrcConf.MAPRED_INPUT_SCHEMA.setString(conf, resultSchemaString)
+
+      if (requestedColIdsOrEmptyFile._1.isEmpty) {
```

Comment:
Thank you for updating, @SaurabhChawla100 . However, `_1` and `_2` is not good in general. You may try like the following at line 182.
```scala
- val requestedColIdsOrEmptyFile =
+ val (requestedColIdsOrEmptyFile, sendActualSchema) =
```

Reply:
done